[BUG] CMake fails while building without documentation on CUDA 11.4

Hi! Super cool library!

The CMake configuration fails when compiling MatX without documentation. It works fine with `-DBUILD_DOCS=ON`. This machine is running CUDA 11.4.

```
➜  MatX git:(main) mkdir build
➜  MatX git:(main) cd build
➜  build git:(main) cmake -DBUILD_TESTS=ON -DBUILD_BENCHMARKS=ON -DBUILD_EXAMPLES=ON -DBUILD_DOCS=OFF ..
-- Auto-detecting GPU architectures since CMAKE_CUDA_ARCHITECTURES not defined
-- The CUDA compiler identification is NVIDIA 11.4.120
-- The CXX compiler identification is GNU 11.1.0
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Check for working CUDA compiler: /opt/cuda/bin/nvcc - skipped
-- Detecting CUDA compile features
-- Detecting CUDA compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/cuda/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Auto detection of gpu-archs: 86
-- Using GPU architectures 86-real
-- Downloading CPM.cmake to /home/luigi/sandbox/MatX/build/cmake/CPM_0.32.2.cmake
-- Setting build type to 'Debug' as none was specified.
-- Found CUDAToolkit: /opt/cuda/include (found version "11.4.120")
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Need libcuda++ 1.6.0 or higher (included in CTK 11.5+). Finding...
CMake Error at build/cmake/CPM_0.32.2.cmake:204 (find_package):
  find_package called with invalid argument "3bc2d4b"
Call Stack (most recent call first):
  build/cmake/CPM_0.32.2.cmake:255 (cpm_find_package)
  build/_deps/rapids-cmake-src/rapids-cmake/cpm/find.cmake:152 (CPMFindPackage)
  cmake/FindLibcudacxx.cmake:21 (rapids_cpm_find)
  cmake/FindLibcudacxx.cmake:44 (find_and_configure_libcudacxx)
  CMakeLists.txt:95 (include)
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,7 +91,7 @@ endif()
 # If we're on CUDA 11.4 or lower we need a newer version of libcudacxx
 if (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11.5)
     message(STATUS "Need libcuda++ 1.6.0 or higher (included in CTK 11.5+). Finding...")
-    set(LIBCUDACXX_VERSION "3bc2d4b" CACHE STRING "Version of libcudacxx to use")
+    set(LIBCUDACXX_VERSION "1.6.0" CACHE STRING "Version of libcudacxx to use")
     include(cmake/FindLibcudacxx.cmake)
     target_include_directories(matx INTERFACE "$<BUILD_INTERFACE:${LIBCUDACXX_INCLUDE_DIR}>")    
 else()

--- a/cmake/FindLibcudacxx.cmake
+++ b/cmake/FindLibcudacxx.cmake
@@ -27,7 +27,7 @@ function(find_and_configure_libcudacxx version)
       ${PROJECT_NAME}-exports
     CPM_ARGS
       GIT_REPOSITORY  https://github.com/NVIDIA/libcudacxx.git
-      GIT_TAG         ${version}
+      GIT_TAG         3bc2d4b
       GIT_SHALLOW     TRUE
       DOWNLOAD_ONLY   TRUE
       OPTIONS         "LIBCXX_INCLUDE_BENCHMARKS OFF"

